File: src/test/java/org/proofit/premiumcalc/MoneyUtilTest.java
package org.proofit.premiumcalc;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import java.math.BigDecimal;

import static org.junit.jupiter.api.Assertions.assertEquals;

class MoneyUtilTest {

    @ParameterizedTest(name = "roundToMoney_{index}: ''{0}'' should be rounded to ''{1}''")
    @CsvSource({
            "-1.234,-1.23",
            "-1.235,-1.24",
            "-1.236,-1.24",
            "1.234,1.23",
            "1.235,1.24",
            "1.236,1.24",
            "2.101,2.10",
            "2.1,2.10",
            "25.1219999999,25.12",
            "123465422534525.1209999,123465422534525.12"
    })
    void roundToMoney(BigDecimal input, BigDecimal expectedOutput) {
        assertEquals(expectedOutput, MoneyUtil.roundToMoney(input));
    }
}

File: src/test/kotlin/org/proofit/premiumcalc/MoneyUtilTest.kt
package org.proofit.premiumcalc

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

import java.math.BigDecimal


class MoneyUtilTest {

    @Test
    fun roundToMoney() {
        assertEquals(BigDecimal("-1.23"), BigDecimal("-1.234").roundToMoney())
        assertEquals(BigDecimal("-1.24"), BigDecimal("-1.235").roundToMoney())
        assertEquals(BigDecimal("-1.24"), BigDecimal("-1.236").roundToMoney())

        assertEquals(BigDecimal("1.23"), BigDecimal("1.234").roundToMoney())
        assertEquals(BigDecimal("1.24"), BigDecimal("1.235").roundToMoney())
        assertEquals(BigDecimal("1.24"), BigDecimal("1.236").roundToMoney())

        assertEquals(BigDecimal("2.10"), BigDecimal("2.101").roundToMoney())
        assertEquals(BigDecimal("2.10"), BigDecimal("2.1").roundToMoney())

        assertEquals(BigDecimal("25.12"), BigDecimal("25.1219999999").roundToMoney())
        assertEquals(BigDecimal("123465422534525.12"), BigDecimal("123465422534525.1209999").roundToMoney())
    }
}
